src/state/InitState.ts

```diff
@@ -144,6 +145,8 @@ export class InitState implements IState{
             application.hedera.mirror.importer.downloader.sources = originalNodeConfiguration.turboNodeProperties.sources;
         }
 
+        application.hedera.mirror.importer.downloader['local']['deleteAfterProcessing'] = !debugMode
+
```

Comment:
Add this to `originalNodeProperties.json` with key `debugNodeProperties` and then use it similar to the others.

Reply:
OK... I did as you asked in the current state of the MR. But I don't get how this solves our problem.

CLIs are not suppose to and can't hold state between command executions. 
Because of this, there is no way of knowing (_inside the debug command_) how was the Local Node started. 

> And we have a requirement to start it, with the flag we are discussing, set to `false`.

That's why I created this function `checkForDebugMode` that checks exactly that. And outputs an error if you try to `debug`  with local node not in debug mode.

We set the flag on `InitState` but we clean it in `CleanUpState`

The only way of passing state to it is config file. As far as I can see.

Reply:
Maybe a good solution would be: